crates/shared/src/event_handling.rs

```diff
@@ -202,6 +233,35 @@ where
     }
 }
 
+fn detect_reorg_path(
+    handled_blocks: &Vec<BlockNumberHash>,
+    current_blocks: &Vec<BlockNumberHash>,
+) -> Result<RangeInclusive<BlockNumberHash>> {
```

Comment:
Since these don't have block hashes, shouldn't this be: `RangeInclusive<u64>`?

Reply:
Indeed. Although there is one place where `Option<H256>` is used - `EmptyPoolInitializer`, I think it is worth removing `Option`. Fixed.